### PR TITLE
Show number of errored samples

### PR DESCRIPTION
**Pull request by KWMORALE**
**Your checklist for this pull request**
- [x] I've read the [contributing guideline](https://github.com/CERT-Polska/mquery/blob/master/CONTRIBUTING.md).
- [x] I've tested my changes by building and running mquery, and testing changed functionality (if applicable)

**What is the current behaviour?**
backend information that the file was not found

**What is the new behaviour?**
Counting number of errored samples using a separate function which in the future may allow it to be placed in more places. Sending this information via redis to the front-end and displaying how many files was missed. The information in the backend still remained.

**Test plan**
index some files and then delete some of them next run some query

**Closing issues**

#101

### CODE REVIEW

File: src/daemon.py [hunk 1]
```diff
@@ -103,11 +104,15 @@ def __execute_yara(self, job: JobId, files: List[str]) -> None:
                     )
             except yara.Error:
                 logging.exception(f"Yara failed to check file {sample}")
+                num_errors += 1
             except FileNotFoundError:
                 logging.exception(
                     f"Failed to open file for yara check: {sample}"
                 )
+                num_errors += 1
+        logging.exception(f"Error files: {num_errors}")
```
Comment (msm-code) [inline src/daemon.py:113]: 1) This is not related to an exception, so `logging.error` at most
2) This is being logged unconditionally (there should be no errors in logs when there are no errors)
3) As a general rule, in case of logging, prefer `logging.func("asdf %s", param)`  instead of `logging.func(f"asdf {param}")`
4) finally, this log is unncessary - all errors are logged already with more details anyway 

```suggestion
        logging.exception(f"Error files: {num_errors}")
```

File: src/mqueryfront/src/QueryResultsStatus.js [hunk 2]
```diff
@@ -220,7 +220,7 @@ class QueryResultsStatus extends Component {
                 </div>
             );
         }
-
+        console.log(this.props.job.files_error);
```
Comment (msm-code) [inline src/mqueryfront/src/QueryResultsStatus.js:223]: unnecessary console log

File: src/daemon.py [hunk 3]
```diff
@@ -103,11 +104,15 @@ def __execute_yara(self, job: JobId, files: List[str]) -> None:
                     )
```
Comment (msm-code) [inline src/daemon.py:104]: move `num_matches` after the `__update_metadata` call, in case it throws an exception

File: src/mqueryfront/src/QueryResultsStatus.js [hunk 4]
```diff
@@ -257,6 +257,12 @@ class QueryResultsStatus extends Component {
                         >
                             {this.props.job.status}
                         </span>
+                        {this.props.job.total_files - lenMatches > 0 && (
```
Comment (msm-code) [inline src/mqueryfront/src/QueryResultsStatus.js:260]: I'd prefer to have the number of errored samples on the progress bar (like the samples in progress, but in red).

**Comment by msm-code** [comment]
(also, please correct the flake8 warning)

**Comment by KWMORALE** [comment]
image
At the moment it looks like this.
In my opinion, it is worth somewhere to place information about how many files were not found not just at the progress bar, in this commit it is next to the status.

File: src/mqueryfront/src/QueryResultsStatus.js [hunk 5]
```diff
@@ -257,6 +268,12 @@ class QueryResultsStatus extends Component {
                         >
                             {this.props.job.status}
                         </span>
+                        {this.props.job.files_errored > 0 && (
+                            <span style={{ color: "grey", fontSize: "12px" }}>
+                                {" "}
+                                {this.props.job.files_errored} files was missed
```
Comment (msm-code) [inline src/mqueryfront/src/QueryResultsStatus.js:274]: `*were` missed.

Or maybe just something like `xxx errors encountered` / `xxx errors during processing`?

Comment (msm-code) [reply src/mqueryfront/src/QueryResultsStatus.js:274]: Also, maybe move it to the right of processing instead?

Also, not sure why but I have this (0 files)

ss

Comment (msm-code) [reply src/mqueryfront/src/QueryResultsStatus.js:274]: Nvm the screenshot, I had the older version.
Still think that maybe it should be moved to the right of `processed`, and maybe displayed in red instead of being gray?

Comment (KWMORALE) [reply src/mqueryfront/src/QueryResultsStatus.js:274]: Red color looked bad there, gray nicely entered the UI.
I was also thinking about putting it there (the right of processed).
